inc/knowbaseitem.class.php

```diff
@@ -1608,76 +1608,93 @@ static function showRecentPopular($type) {
 
       $faq = !Session::haveRight(self::$rightname, READ);
 
+      $criteria = [
+         'SELECT'    => ['glpi_knowbaseitems.*'],
+         'DISTINCT'  => true,
+         'FROM'      => self::getTable(),
+         'WHERE'     => [],
+         'LIMIT'     => 10
+      ];
+
       if ($type == "recent") {
-         $orderby = "ORDER BY `date` DESC";
+         $criteria['ORDERBY'] = 'date DESC';
          $title   = __('Recent entries');
       } else if ($type == 'lastupdate') {
-         $orderby = "ORDER BY `date_mod` DESC";
+         $criteria['ORDERBY'] = 'date_mod DESC';
          $title   = __('Last updated entries');
       } else {
-         $orderby = "ORDER BY `view` DESC";
+         $criteria['ORDERBY'] = 'view DESC';
          $title   = __('Most popular questions');
       }
 
-      $faq_limit = "";
-      $addselect = "";
       // Force all joins for not published to verify no visibility set
-      $join = self::addVisibilityJoins(true);
+      $restrict = self::getVisibilityCriteria(true);
+      unset($restrict['WHERE']);
+      unset($restrict['SELECT']);
+      $criteria = array_merge($criteria, $restrict);
 
       if (Session::getLoginUserID()) {
-         $restrict = self::addVisibilityRestrict();
-         $faq_limit .= "WHERE ".$DB->mergeStatementWithParams(
-            $restrict['sql'],
-            $restrict['params']
-         );
+         $restrict = self::getVisibilityCriteria();
+         $criteria['WHERE'] = array_merge($criteria['WHERE'], $restrict['WHERE']);
       } else {
          // Anonymous access
          if (Session::isMultiEntitiesMode()) {
-            $faq_limit .= " WHERE (`glpi_entities_knowbaseitems`.`entities_id` = 0
-                                   AND `glpi_entities_knowbaseitems`.`is_recursive` = 1)";
-         } else {
-            $faq_limit .= " WHERE 1";
+            $criteria['WHERE']['glpi_entities_knowbaseitems.entities_id'] = 0;
+            $criteria['WHERE']['glpi_entities_knowbaseitems.is_recursive'] = 1;
          }
       }
 
       // Only published
-      $faq_limit .= " AND (`glpi_entities_knowbaseitems`.`entities_id` IS NOT NULL
-                           OR `glpi_knowbaseitems_profiles`.`profiles_id` IS NOT NULL
-                           OR `glpi_groups_knowbaseitems`.`groups_id` IS NOT NULL
-                           OR `glpi_knowbaseitems_users`.`users_id` IS NOT NULL)";
+      $criteria['WHERE'][] = [
+         'OR'  => [
+            'NOT' => [
```

Comment:
This will generate a `NOT(a AND b AND c AND d)` instead of a `NOT(a) OR NOT(b) ...`.